internal/broker/config.go

```diff
@@ -53,6 +96,45 @@ func getDropInFiles(cfgPath string) ([]any, error) {
 	return dropInFiles, nil
 }
 
+func populateUsersConfig(cfg *userConfig, users *ini.Section) {
+	if users == nil {
+		// The default behavior is to allow only the owner
+		cfg.ownerAllowed = true
+		cfg.firstUserBecomesOwner = true
+		return
+	}
+
+	cfg.homeBaseDir = users.Key(homeDirKey).String()
+	cfg.allowedSSHSuffixes = strings.Split(users.Key(sshSuffixesKey).String(), ",")
+
+	if cfg.allowedUsers == nil {
+		cfg.allowedUsers = make(map[string]bool)
+	}
+
+	allowedUsers := users.Key(allowedUsersKey).Strings(",")
+	if len(allowedUsers) == 0 {
+		allowedUsers = append(allowedUsers, OwnerUserKeyword)
+	}
+
+	for _, user := range allowedUsers {
+		if user == AllUsersKeyword {
+			cfg.allUsersAllowed = true
+		} else if user == OwnerUserKeyword {
+			cfg.ownerAllowed = true
+			if !users.HasKey(ownerKey) {
+				// If owner is unset, then the first user becomes owner
+				cfg.firstUserBecomesOwner = true
+			}
+		} else {
+			cfg.allowedUsers[user] = true
+		}
+	}
+
+	// We need to read the owner key before we call HasKey, because the key is created
+	// after we call the "Key" function and we can't distinguish between empty and unset.
+	cfg.owner = users.Key(ownerKey).String()
```

Comment:
I don't understand this comment.

> We need to read the owner key before we call HasKey

`users.HasKey(ownerKey)` is called above, so we don't read the owner key before we call `HasKey`?

> because the key is created after we call the "Key"

What does "created after" mean? Is it created *when* we call the `Key` method or what causes it to be created?

Reply:
yes, https://github.com/go-ini/ini/blob/main/section.go#L171

Reply:
Thanks for updating the comment, it makes more sense now :)